### PR TITLE
OCPBUGS#5244: When an admin user associates a floating IP to the reservation port, delete operation hangs

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.12+

Issue: https://issues.redhat.com/browse/OCPBUGS-5244

Link to docs preview: https://58176--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-public-cloud-platform-considerations_egress-ips

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by rlobillo** [comment]
@andreaskaris If I am understanding correct, the scenario you are mentioning is only applicable for 4.12. Thanks to below 4.13 epic: 
https://issues.redhat.com/browse/OSASINFRA-3007

The port is not recreated on an failover scenario. Do you think there are other scenarios that can lead to the problem described in https://issues.redhat.com/browse/OCPBUGS-5244 ?

Thanks.

**Comment by dulek** [comment]
@rlobillo: This is a different issue from the one you think about. This one is when it's admin that created the FIP and assigned it to the port. Then the cloud-network-config-controller wouldn't be able to detach it from the port. It still shouldn't be a 500 error from Neutron, but in general it's not something we support, operations have to be done from the same tenant.

**Comment by dulek** [comment]
@xenolinux: Current version has little sense to me, @andreaskaris explained it better a few comments above.

Also please pay attention that you've edited this *inside* the `ifdef::openshift-sdn[]` block. The problem you're trying to describe happens with both openshift-sdn and ovn-kubernetes. The problem with lost floating IP assignment on failover currently only happens with openshift-sdn, that's why we have the ifdef block.

**Comment by xenolinux** [comment]
Hi @andreaskaris I updated the PR. Moved the text out of the `ifdef::openshift-sdn[]` block since this scenario is valid for both openshift-sdn and ovn-kubernetes. Please let me know if any other change is required.

File: modules/nw-egress-ips-about.adoc [hunk 1]
```diff
@@ -88,6 +88,11 @@ The {rh-openstack} egress IP address feature creates a Neutron reservation port
 ====
 endif::openshift-sdn[]
 
+[NOTE]
+====
+When a cluster administrator assigns a floating IP to the reservation port, {product-title} cannot delete the reservation port. The `CloudPrivateIPConfig` object cannot perform delete and move operations until the floating IP is unassigned from the reservation port.
```
Comment (dulek) [inline modules/nw-egress-ips-about.adoc:93]: Can we make it more specific that this is about OpenStack cluster administrator, not OpenShift? Right now this is ambiguous.

I think we can use `{rh-openstack}` to achieve that.

**Review by dulek — CHANGES_REQUESTED** [review]
So I'm not 100% sure move EgressIP operation is affected, but it's totally fine to play it safe and add it. In general the bottom line of this note is - don't assign FIPs using admin accounts.

Tiny remark inline. Besides that this is good to go.

File: modules/nw-egress-ips-about.adoc (hunk 1 above)
Comment (xenolinux) [reply modules/nw-egress-ips-about.adoc]: @dulek Addressed.

Comment (xenolinux) [reply modules/nw-egress-ips-about.adoc]: Does the note look good now?

Comment (dulek) [reply modules/nw-egress-ips-about.adoc]: Looks good from my perspective.

**Review by abhatt-rh — COMMENTED** [review]
LGTM, except a minor comment!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**